spec/services/hashtag_autocomplete_service_spec.rb

```diff
@@ -92,7 +97,9 @@ def self.search(guardian_scoped, term, limit)
     end
 
     it "does not search other data sources if the limit is reached by earlier type data sources" do
-      Site.any_instance.expects(:categories).never
+      # only expected once to try get the exact matches first
+      site_guardian_categories = Site.new(guardian).categories
+      Site.any_instance.expects(:categories).once.returns(site_guardian_categories)
```

Comment:
Do we have only reason to think it could search twice? Feels like we are testing for something that can't happen.

If I read this correctly we are now fetching once for each data source as the ordering is on all types now.

So now we search each type with the site setting as limit and order by best match, are we doing more requests if after this first pass we are still not at the site setting limit?

Reply:
Yes it shouldn't search twice, because as we find results from the first type we subtract the count of those results from the limit for subsequent searches, and if we've already hit the limit then we don't continue searching